src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -142,22 +149,14 @@ public Task InitializeAsync()
             return Task.CompletedTask;
         }
 
-        public string GraphQLSchema()
+        private string GraphQLSchema()
         {
-            if (_cosmosDb.GraphQLSchema is null && _fileSystem.File.Exists(_cosmosDb.GraphQLSchemaPath))
-            {
-                _cosmosDb = _cosmosDb with { GraphQLSchema = _fileSystem.File.ReadAllText(_cosmosDb.GraphQLSchemaPath) };
-            }
-
-            if (_cosmosDb.GraphQLSchema is null)
+            if (_cosmosDb.GraphQLSchema is not null)
             {
-                throw new DataApiBuilderException(
-                    "GraphQL Schema isn't set.",
-                    System.Net.HttpStatusCode.InternalServerError,
-                    DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                return _cosmosDb.GraphQLSchema;
             }
 
-            return _cosmosDb.GraphQLSchema;
+            return _fileSystem.File.ReadAllText(_cosmosDb.Schema!);
```

Comment:
will the file specified in the `schema` always exist?

Reply:
With the new code, it will miss the case when GraphQLSchema input from the HttpEndpoint is null, we should be erroring out in that case as well.

Reply:
Yes, but the `RuntimeConfigProvider` which does the initialisation will validate that a schema was provided:

```csharp
    private static RuntimeConfig HandleCosmosNoSqlConfiguration(string? schema, RuntimeConfig runtimeConfig, string connectionString)
    {
        DbConnectionStringBuilder dbConnectionStringBuilder = new()
        {
            ConnectionString = connectionString
        };

        if (string.IsNullOrEmpty(schema))
        {
            throw new ArgumentException($"'{nameof(schema)}' cannot be null or empty.", nameof(schema));
        }
```

So it won't start if you don't provide that anyway.